### PR TITLE
feat(L2): add SP1 mock prover

### PR DESCRIPTION
**Pull request by ManuelBilbao**
**Motivation**

Add SP1 mock prover so it's easy later to integrate the real one.

**Description**

The prover generates a proof for the nth (the ID) fibonacci number. It should be easy later to integrate with a real ELF

### CODE REVIEW

**Comment by unbalancedparentheses** [comment]
I hate mocks. What is this @jrchatruc xD

File: crates/l2/prover/sp1/elf/riscv32im-succinct-zkvm-elf
Comment (ilitteri) [inline crates/l2/prover/sp1/elf/riscv32im-succinct-zkvm-elf:1]: Do we want this build to be tracked?

Comment (ManuelBilbao) [reply crates/l2/prover/sp1/elf/riscv32im-succinct-zkvm-elf:1]: This is the Fibonacci ELF that is used currently as the program to prove. If we don't want it to be in the repo, we could download it from somewhere else at setup time, but in any case we need it (see https://github.com/lambdaclass/ethereum_rust/blob/be53c79b7ce595bd7ed7e9b71445d7eed9134131/crates/l2/prover/sp1_prover.rs#L6)

Comment (ilitteri) [reply crates/l2/prover/sp1/elf/riscv32im-succinct-zkvm-elf:1]: Let's double-check it cannot be built running `cargo prove build` inside the program's directory.

**Review by ilitteri — APPROVED** [review]

**Pull request merged**